web/src/main/java/org/mskcc/oncotree/api/TumorTypesApi.java

```diff
@@ -3,170 +3,90 @@
-        @RequestParam(value = "version", required = false) String version,
-        @ApiParam(value = "The flat list of tumor types", defaultValue = "false")
-        @RequestParam(value = "flat", required = false, defaultValue = "false") Boolean flat,
-        @ApiParam(value = "Indicator that whether should include deprecated tumor types.", defaultValue = "false")
-        @RequestParam(value = "deprecated", required = false, defaultValue = "false") Boolean deprecated
-//        , @ApiParam(value = "The callback function name. This has to be used with dataType JSONP.")
-//        @RequestParam(value = "callback", required = false) String callback
-    )
-        throws NotFoundException {
-        InlineResponse200 response200 = new InlineResponse200();
-        Meta meta = new Meta();
-        meta.setCode(200);
-        response200.setMeta(meta);
-
-        Map<String, TumorType> tumorTypes = new HashMap<>();
-
+        @RequestParam(value = "version", required = false) String version
+    ) {
         Version v = (version == null) ? VersionUtil.getDefaultVersion() : VersionUtil.getVersion(version);
-
-        tumorTypes = CacheUtil.getOrResetTumorTypesByVersion(v);
-
-        if (flat) {
-            response200.setData(TumorTypesUtil.flattenTumorTypes(tumorTypes, null));
-        } else {
-            response200.setData(tumorTypes);
-        }
-        return new ResponseEntity<InlineResponse200>(response200, HttpStatus.OK);
-    }
-
-
-    //    @ApiOperation(value = "Return the selected tumor type children list.", notes = "Return the selected tumor type children list. By default, it will only return the children IDs.", response = ChildrenListResp.class)
-//    @io.swagger.annotations.ApiResponses(value = {
-//        @io.swagger.annotations.ApiResponse(code = 200, message = "OK")})
-//    @RequestMapping(value = "/{id}/children",
-//        produces = {"application/json"},
-//        method = RequestMethod.GET)
-    public ResponseEntity<ChildrenListResp> tumorTypesIdChildrenGet(
-        @ApiParam(value = "The numerical ID of the desired tumor type", required = true)
-        @PathVariable("id") Integer id
-//        , @ApiParam(value = "The callback function name. This has to be used with dataType JSONP.")
-//        @RequestParam(value = "callback", required = false) String callback
-
-
-    )
-        throws NotFoundException {
-        // do some magic!
-        return new ResponseEntity<ChildrenListResp>(HttpStatus.OK);
+        Map<String, TumorType> tumorTypes = CacheUtil.getTumorTypesByVersion(v);
+        logger.debug("tumorTypesGet() -- returning " + tumorTypes.size() + " tumor types");
+        return tumorTypes;
     }
 
+    @ApiOperation(value = "Return all available tumor types as a list.", notes = "", response = TumorType.class)
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Tumor Types list.")})
+    @RequestMapping(value = "/flat",
```

Comment:
Does this mean we are dropping the parameter ?flat={true/false} ?  (splitting it into a separate endpoint)
I think that's ok, but it raises the question of the other endpoints ... should our responses to /search have a /search/flat endpoint too (to leave room for the non-flat version one level up) ?